generate/generate.js

```diff
@@ -42,26 +42,26 @@ const generate = async () => {
 
   await exec('yarn truffle-compile');
 
-  const bytecodes = tokens.reduce((output, { name }) => {
+  const bytecodes = tokens.reduce((output, { name, currencyId }) => {
     const { deployedBytecode } = require(`../build/contracts/${name}ERC20.json`);
-    return [...output, [name, deployedBytecode]];
+    return [...output, [name, "0x" + (2048 + Number(currencyId)).toString(16).padStart(40,0), deployedBytecode]];
```

Comment:
make a `const ERC20_PREDEPLOY_ADDRESS_START = 2048` at top of the file to mark this clear